templates/new/web_dodo.py.txt

```diff
@@ -246,6 +252,357 @@ def task_web_local():
+
+    # Construct build command
+    image_uri, build_cmd = compute_docker_build_cmd("Dockerfile-compute-gpu", True)
+
+    # Construct push command
+    push_cmd = "docker push %s" % image_uri
+
+    return {{
+        'basename': 'buildComputeGPU',
+        'actions': [compute_docker_login, build_cmd, 'rm credentials.json', push_cmd]
+    }}
+
+def compute_check_image(is_gpu=False):
+    """Checks whether there has been an image build for compute"""
+
+    image_name = IMAGE if not is_gpu else ("%s-gpu" % IMAGE)
+    process = subprocess.Popen(
+        [
+            'docker',
+            'images',
+            '-q',
+            '-f',
+            'reference=gcr.io/*/%s' % image_name
+        ], stdout=subprocess.PIPE)
+    process.wait()
+
+    if not process.stdout.read():
+        print("No compute image found!")
+        print("Make sure you run the following before deploying to the cloud!")
+        print("$ surround run", "buildComputeGPU" if is_gpu else "buildCompute")
+        return False
+
+    return True
+
+def compute_get_ml_client(credentials):
+    """Gets the ML API client from Google using the credentials.
+    Tries multiple times due to random Google bug"""
+
+    count = 0
+    cloud_ml = None
+    while count < 5:
+        try:
+            # python api for AI Platform
+            cloud_ml = discovery.build('ml', 'v1', credentials=credentials)
+        except ssl.SSLEOFError:
+            count += 1
+            continue
+
+        break
+
+    if not cloud_ml:
+        print("Failed to create ML client, try again later!")
+
+    return cloud_ml
+
+def compute_deploy_job(mode="batch", is_gpu=False):
+    """Deploys a compute job to Google Cloud (AI Platform)"""
+
+    credentials = compute_get_credentials()
+    if not credentials:
+        return False
+
+    cloud_ml = compute_get_ml_client(credentials)
+    if not cloud_ml:
+        return False
+
+    # get current time stamp
+    now = datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    # Generate configuration
+    project_id = 'projects/%s' % credentials.project_id
+    job_name = '{project_name}_%s_%s_%s' % (mode, ("gpu" if is_gpu else "cpu"), now)
+    image_name = IMAGE if not is_gpu else ("%s-gpu" % IMAGE)
+    image_uri = "gcr.io/%s/%s" % (credentials.project_id, image_name)
+
+    # Only run as experiment if using cloud storage for experiments
+    global_config = get_surround_config()
+    run_as_experiment = not os.path.exists(global_config.get_path("experiment.url"))
+
+    job_spec = {{
+        'jobId': job_name,
+        'trainingInput': {{
+            "scaleTier": CONFIG["compute"]["gpu" if is_gpu else "cpu"]["scale-tier"],
+            "masterConfig": {{
+                "imageUri": image_uri
+            }},
+            "args": [
+                '--mode=%s' % mode,
+                '--runner=1'
+            ],
+            "region": CONFIG["compute"]["gpu" if is_gpu else "cpu"]["region"]
+        }}
+    }}
```

Comment:
Google is updating AI Platform, in a year this training input will fail. Need to add “runtimeVersion” and “pythonVersion” parameters. 

> Starting on January 13, 2020, “runtimeVersion” and “pythonVersion” will become required parameters when creating a new Job or Version. Previously, these were optional and defaulted to versions 1.0 and 2.7. Please take into consideration that any API calls or gcloud commands that do not specify these new fields will fail.

See an example of doing this [here](https://cloud.google.com/ml-engine/docs/versioning).

Reply:
Note: if you provide the master Docker image uri (as you have done here) then you do not need to include the `runtimeVersion` parameter.

Reply:
Tested today (4th of Feb), fails when specifying a `pythonVersion` and works when leaving the code as it is, so I think this change only effects people using Google's runtimes rather than custom images